conformance/tests/gatewayclass-observed-generation-bump.yaml

```diff
@@ -0,0 +1,7 @@
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: GatewayClass
+metadata:
+  name: gatewayclass-observed-generation-bump
+spec:
+  controllerName: "{GATEWAY_CONTROLLER_NAME}"
```

Comment:
This seems like a significant change. As I understand it, this requires implementations to watch GatewayClasses matching their controller name and then update status on those. I think in some cases GatewayClass is part of the deployment model and each controller supports exactly one class. I could be wrong, but need to get some more input from other implementations before moving forward with this.

Reply:
I can `t.Skip()` this test for now as not to hold the entire PR

Reply:
I put it behind a support feature flag for now